crates/forge/src/running.rs

```diff
@@ -70,25 +71,17 @@ fn build_hints_dict<'b>(
     (hints_dict, string_to_hint)
 }
 
-pub(crate) async fn blocking_run_from_test(
+pub(crate) fn blocking_run_from_test(
     args: Vec<Felt252>,
     case: Arc<TestCase>,
     runner: Arc<SierraCasmRunner>,
     runner_config: Arc<RunnerConfig>,
     runner_params: Arc<RunnerParams>,
-    sender: Option<Sender<()>>,
-) -> Result<TestCaseSummary> {
+    send: Sender<()>,
+) -> JoinHandle<Result<TestCaseSummary>> {
     tokio::task::spawn_blocking(move || {
-        run_test_case(
-            args,
-            &case,
-            &runner,
-            &runner_config,
-            &runner_params,
-            &sender,
-        )
+        run_test_case(args, &case, &runner, &runner_config, &runner_params, &send)
```

Comment:
Do we need to pass the sender down to the blocking task?

Reply:
I can refactor it a little bit